addons/pos_gift_card/static/src/js/GiftCardPopup.js

```diff
@@ -52,6 +80,7 @@ odoo.define("pos_gift_card.GiftCardPopup", function (require) {
 
     async getGiftCard() {
       if (this.state.giftCardBarcode == "") return;
+      this.state.giftCardBarcode = this.state.giftCardBarcode.trim();
```

Comment:
This is inside a getter. It's not good to modify the state at this point, because it will trigger rerender. What you can do is save the trimmed value in a variable and use that in the comparison part.